src/main/java/filter/Parser.java

```diff
@@ -181,6 +181,8 @@ private FilterExpression parseSeparatedContent(Supplier<FilterExpression> conten
     private FilterExpression parseQualifierContent(QualifierType type) {
         if (type == QualifierType.SORT) {
             return parseSortKeys();
+        } else if (isCompoundId(type, lookAhead())) {
```

Comment:
Checking for failure here leads to a non-id qualifier being used with a compound id being a parse error. Could we follow the other `parse*` functions and let it fail later, as a semantic error? If for some reason we want to use compound ids with other qualifiers.